odoo/orm/fields_misc.py

```diff
@@ -93,6 +93,7 @@ def convert_to_export(self, value, record):
 
 class Id(Field[IdType | typing.Literal[False]]):
     """ Special case for field 'id'. """
+    # Note: This field type is not necessarily an integer!
     type = 'integer'  # note this conflicts with Integer
```

Comment:
Why do we need an `Id` field? Why not just use `Integer` by default?

Reply:
Because `Id` bypasses the cache and `Integer` does not.